### PR TITLE
Convert IRI -> CURIE using `bioregistry` while `parse`-ing a KGCL command.

### PR DESCRIPTION
**Pull request by hrshdhgd**
Triggered by [this comment](https://github.com/INCATools/ontology-access-kit/issues/208#issuecomment-1192925480).

 - [x] Add bio registry dependency
 - [x] Convert IRI -> CURIE using `curie_from_iri`
 - [x] Remove `<>` around URIs : fixes https://github.com/INCATools/kgcl-rdflib/issues/58
 - [x] Remove quotes around labels: fixes https://github.com/INCATools/kgcl-rdflib/issues/43

### CODE REVIEW

File: src/kgcl_schema/grammar/parser.py [hunk 1]
```diff
@@ -54,6 +58,15 @@ def parse_statement(input: str) -> Change:
 
     Return an instantiated dataclass object from model.kgcl_schema.
     """
+    regex = r'<http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
+    uri = re.findall(regex, input)
+    if uri:
+        # curie = curie_from_iri(uri[0].replace("<", "").replace(">",""))
+        pref, i = parse_iri(uri[0].replace("<", "").replace(">",""))
```
Comment (matentzn) [inline src/kgcl_schema/grammar/parser.py:65]: There it is again - I am having long debates about the `<>` with @joeflack4 as well. This URI hacking smells bad, like the input is not properly sanitised. Why is this necessary, where does the `<>` come from?

Comment (matentzn) [reply src/kgcl_schema/grammar/parser.py:65]: Same with the regex above. you seem to _expect_ uris with `<>`..

File: src/kgcl_schema/grammar/parser.py [hunk 2]
```diff
@@ -54,6 +58,15 @@ def parse_statement(input: str) -> Change:
...
+        pref = get_preferred_prefix(pref)
+        curie = curie_to_str(pref, i)
+        input = input.replace(uri[0], curie)
```
Comment (matentzn) [inline src/kgcl_schema/grammar/parser.py:68]: Beter use https://github.com/cthoyt/curies

File: src/kgcl_schema/grammar/parser.py [hunk 3]
```diff
@@ -54,6 +58,15 @@ def parse_statement(input: str) -> Change:
 
     Return an instantiated dataclass object from model.kgcl_schema.
     """
+    regex = r'<http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
```
Comment (matentzn) [inline src/kgcl_schema/grammar/parser.py:61]: This does not seem to end with `>`, but see comments below.

File: src/kgcl_schema/grammar/parser.py [hunk 4]
```diff
@@ -54,6 +58,15 @@ def parse_statement(input: str) -> Change:
...
+    uri = re.findall(regex, input)
```
Comment (matentzn) [inline src/kgcl_schema/grammar/parser.py:62]: If the input string has multiple URI matches, only the first one will be dealt with

File: src/kgcl_schema/grammar/parser.py [hunk 5]
```diff
@@ -54,6 +58,15 @@ def parse_statement(input: str) -> Change:
...
+    if uri:
```
Comment (matentzn) [inline src/kgcl_schema/grammar/parser.py:63]: Shouldnt there be an `else` with that if to sanitise if the input string is not found to be a URI?

File: src/kgcl_schema/grammar/parser.py (hunk 2 above)
Comment (matentzn) [reply src/kgcl_schema/grammar/parser.py:68]: (but dont sweat it)

File: src/kgcl_schema/grammar/parser.py [hunk 6]
```diff
@@ -54,6 +58,15 @@ def parse_statement(input: str) -> Change:
...
+        # curie = curie_from_iri(uri[0].replace("<", "").replace(">",""))
+        pref, i = parse_iri(uri[0].replace("<", "").replace(">",""))
+        pref = get_preferred_prefix(pref)
+        curie = curie_to_str(pref, i)
+        input = input.replace(uri[0], curie)
+
```
Comment (matentzn) [inline src/kgcl_schema/grammar/parser.py:69]: It seems very hacky to replace URIs with CURIEs just so the KGCL parser works. Why is that necessary?

File: src/kgcl_schema/grammar/parser.py (hunk 1 above)
Comment (hrshdhgd) [reply src/kgcl_schema/grammar/parser.py:65]: That's what is provided in the command itself. I think there is documentation about adding the `<>` for URIs

File: src/kgcl_schema/grammar/parser.py (hunk 4 above)
Comment (hrshdhgd) [reply src/kgcl_schema/grammar/parser.py:62]: I'm working with just adding a synonym as of now. Would there be a command like:

```
create synonym 'ABCD3' for http://purl.obolibrary.org/obo/FBbt_00004751 http://purl.obolibrary.org/obo/FBbt_1234
```

File: src/kgcl_schema/grammar/parser.py (hunk 6 above)
Comment (hrshdhgd) [reply src/kgcl_schema/grammar/parser.py:69]: When adding synonyms, apparently only CURIEs are expected and ehnce the conversion from URI => CURIE. Just a judgement call, there could be a better way of handling this. Ideas?

File: src/kgcl_schema/grammar/parser.py (hunk 4 above)
Comment (matentzn) [reply src/kgcl_schema/grammar/parser.py:62]: no but there can easily be a statement like:

create edge http://purl.obolibrary.org/obo/FBbt_00004751 to http://purl.obolibrary.org/obo/FBbt_1234

File: src/kgcl_schema/grammar/parser.py (hunk 6 above)
Comment (matentzn) [reply src/kgcl_schema/grammar/parser.py:69]: I dont know how that parser works, but if CURIES are expected, wouldn't it be better to make sure the input is "valid kgcl" using a validator and not trying to hack anything? The user can be forced to provide CURIEs if that is what the spec wants..

In any case, I would for sure factor out the method `def replace_uris_with_curies_in_kgcl_command(input)` into its how, document it etc. Not convinced this is great but if its necessary to be parseable than so be it.

File: src/kgcl_schema/grammar/parser.py (hunk 5 above)
Comment (hrshdhgd) [reply src/kgcl_schema/grammar/parser.py:63]: No .. if it's not a URI the assumption is it must be a CURIE already, hence no extra steps needed.

Comment (matentzn) [reply src/kgcl_schema/grammar/parser.py:63]: Yeah ok, I think I got it now.

File: src/kgcl_schema/grammar/parser.py (hunk 1 above)
Comment (hrshdhgd) [reply src/kgcl_schema/grammar/parser.py]: Addressed this. Removed all `<>` references for URIs in KGCL.

File: src/kgcl_schema/grammar/parser.py (hunk 3 above)
Comment (hrshdhgd) [reply src/kgcl_schema/grammar/parser.py]: Resolved.

File: src/kgcl_schema/grammar/parser.py (hunk 4 above)
Comment (hrshdhgd) [reply src/kgcl_schema/grammar/parser.py]: Addressed

**Review by cmungall — APPROVED** [review]
